src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/InternalCalls.cs

```diff
@@ -158,8 +156,8 @@ internal static int RhEndNoGCRegion()
         [RuntimeImport(Redhawk.BaseName, "RhpGcSafeZeroMemory")]
         internal static extern unsafe ref byte RhpGcSafeZeroMemory(ref byte dmem, nuint size);
 
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        [RuntimeImport(Redhawk.BaseName, "memmove")]
+        [DllImport(Redhawk.BaseName)]
+        [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
```

Comment:
The extra PInvoke transitions can regress the perf. Can it be deleted and replaced with Unsafe.CopyBlock, like what we have done here https://github.com/dotnet/runtime/blob/c6906ae60861633919c411e6e76e683bab3c6ecd/src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs#L295 ?

(Or just drop this change from this PR.)

Reply:
I’ll check. Worst case I can wrap it under `#if TARGET_X86` if I don’t find a better solution.

Reply:
I think I initially dismissed this concern because it's only used in `Buffer.NativeAot.cs` where the CoreCLR counterpart uses QCalls. The source code in the shared `Buffer.cs` also mentions that it's supposed to be QCalls in slow path.